### PR TITLE
Fix building error at backtrace

### PR DESCRIPTION
**Pull request by BaoshanPang**
I get this error when builind backtrace:

error[E0308]: mismatched types
  --> /folk/prj-rust-dev/bpang/ala-diab-pb19l/fix_bugs/latest/.cargo/git/checkouts/backtrace-rs-fb1f822361417489/e745627/src/symboli\
ze/dladdr.rs:38:73
   |
38 |                         Some(SymbolName::new(slice::from_raw_parts(ptr, len)))
   |                                                                         ^^^ expected usize, found u64
help: you can convert an `u64` to `usize` and panic if the converted value wouldn't fit

It shows the 'size_t' is expected to be defined as "usize".

### CODE REVIEW

**Comment by gnzlbg** [comment]
There is no libc function in that error message.

File: src/vxworks/mod.rs [hunk 1]
```diff
@@ -25,23 +25,11 @@ pub type c_ulonglong = u64;
...
-        pub type size_t = c_ulonglong;
-        pub type ssize_t = c_longlong;
-        pub type ptrdiff_t = c_longlong;
-        pub type intptr_t = c_longlong;
-        pub type uintptr_t = c_ulonglong;
-    } else {
-        // Unknown target_pointer_width
-    }
-}
+pub type size_t = usize;
```
Comment (gnzlbg) [inline src/vxworks/mod.rs]: How is `size_t` defined in the VxWorks headers ? Can you copy that part of the header ?

**Comment by gnzlbg** [comment]
Depending on how the things are defined in the C header, this line might have a bug:

> https://github.com/rust-lang/backtrace-rs/blob/master/src/symbolize/dladdr.rs#L38

Instead of `len`, it should be `len as _`.

**Comment by BaoshanPang** [comment]
> There is no libc function in that error message.

Sorry, I should paste more code:

                 unsafe {
                        let len = libc::strlen(self.inner.dli_sname);
                        Some(SymbolName::new(slice::from_raw_parts(ptr, len)))
                    }

File: src/vxworks/mod.rs (hunk 1 above)
Comment (BaoshanPang) [reply src/vxworks/mod.rs]: It is defined like this:
#ifdef _WRS_CONFIG_LP64
typedef unsigned long long size_t;
#else
typedef unsigned int size_t;
#endif

Comment (gnzlbg) [reply src/vxworks/mod.rs]: How are uintptr_t and intptr_t defined ?

**Comment by BaoshanPang** [comment]
> Depending on how the things are defined in the C header, this line might have a bug:
> 
> > https://github.com/rust-lang/backtrace-rs/blob/master/src/symbolize/dladdr.rs#L38
> 
> Instead of `len`, it might need to be `len as _`. Hard to tell without seeing the headers.


I see the size_t is defined as usize almost for every targets:

$ grep "pub type size_t" * -rn
cloudabi/mod.rs:14:pub type size_t = usize;
fuchsia/mod.rs:23:pub type size_t = usize;
hermit/mod.rs:29:pub type size_t = usize;
sgx.rs:16:pub type size_t = usize;
switch.rs:16:pub type size_t = usize;
unix/mod.rs:19:pub type size_t = usize;
vxworks/mod.rs:28:pub type size_t = usize;
wasi.rs:16:pub type size_t = usize;
windows/mod.rs:16:pub type size_t = usize;

File: src/vxworks/mod.rs (hunk 1 above)
Comment (BaoshanPang) [reply src/vxworks/mod.rs]: #ifdef _WRS_CONFIG_LP64
typedef long long intptr_t;
typedef unsigned long long uintptr_t;
#else
typedef int intptr_t;
typedef unsigned int uintptr_t;
#endif

Comment (gnzlbg) [reply src/vxworks/mod.rs]: So I suggest defining `uintptr_t = usize` and `intptr_t = isize`, and using `uintptr_t` to define `size_t`, and `intptr_t` to define `ssize_t`.

Comment (BaoshanPang) [reply src/vxworks/mod.rs]: How about ptrdiff_t?

Comment (gnzlbg) [reply src/vxworks/mod.rs]: It's ok to make it `isize` I think.

**Comment by BaoshanPang** [comment]
It seems the fail is caused by some network issue.

>>E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?

**Pull request merged**